File: src/main/java/com/wanted/preonboarding/ticket/domain/PerformanceSeatInfo.java
package com.wanted.preonboarding.ticket.domain;

import jakarta.persistence.*;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedDate;

import java.time.LocalDateTime;

@Getter
@Entity
@Table(name = "performance_seat_info")
@EqualsAndHashCode(onlyExplicitlyIncluded = true)
public class PerformanceSeatInfo {
    @Id
    @Column(name = "id")
    @EqualsAndHashCode.Include
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "performance_id")
    private Performance performance;
    
    @Column(name = "round")
    private int round;
    
    @Column(name = "gate")
    private int gate;
    
    @Column(name = "line")
    private String line;
    
    @Column(name = "seat")
    private int seat;
    
    @Column(name = "is_reserve")
    private String isReserve;

    @CreatedDate
    @Column(name = "created_at")
    private LocalDateTime createdAt;

    @LastModifiedDate
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;
}

File: src/main/kotlin/com/wanted/preonboarding/ticket/domain/PerformanceSeatInfo.kt
package com.wanted.preonboarding.ticket.domain

import java.time.LocalDateTime

class PerformanceSeatInfo(
    private val id: PerformanceSeatInfoId = PerformanceSeatInfoId(0),
    private val seatInfo: SeatInfo,
    private var isReserve: Boolean,
    private val createdAt: LocalDateTime,
    private val updatedAt: LocalDateTime? = null,
) {
    fun isSameSeat(seatInfo: SeatInfo): Boolean {
        return this.seatInfo == seatInfo
    }

    fun isReserveAvailable(): Boolean {
        return isReserve
    }

    fun setReserveAvailable() {
        isReserve = false
    }

    fun setReserveNotAvailable() {
        isReserve = true
    }
}
